### PR TITLE
Support MySQL LATERAL query parse

### PR DESCRIPTION
**Pull request by zihaoAK47**
Ref #25535.

Changes proposed in this pull request:
  - Support MySQL LATERAL query parse
  - Add test sql

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ✓] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ✓] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ✓] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ✓] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by zihaoAK47** [comment]
@strongduanmu Hi mentor please review

https://github.com/mysql/mysql-server/blob/8.0/sql/sql_yacc.yy#L11842
https://github.com/mysql/mysql-server/blob/8.0/sql/sql_yacc.yy#L12037
https://github.com/mysql/mysql-server/blob/8.0/sql/sql_yacc.yy#L12075

File: parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4 [hunk 1]
```diff
@@ -287,7 +287,10 @@ tableReference
     ;
 
 tableFactor
-    : tableName partitionNames? (AS? alias)? indexHintList? | subquery AS? alias (LP_ columnNames RP_)? | LP_ tableReferences RP_
```
Comment (strongduanmu) [inline parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: Do you think we should refactor antlr parse logic accord to MySQL yacc?

**Review by strongduanmu — APPROVED** [review]
This pr is great, I will merge it.

File: parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4 (hunk 1 above)
Comment (zihaoAK47) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: Aha, it was only after going through all the rules of table_factor in MySQL yacc that I finally understood how to modify it. If want to have a better reading experience and easily refer back to MySQL yacc for future modifications, I believe refactoring is necessary. What do you think, mentor? If we need to refactor, I will start immediately.

Comment (strongduanmu) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: I think this priority can be slightly lower, you can try to handle the unsupported sql parsing first.

Comment (strongduanmu) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: In addition, can you send your contact information to my email address? @zihaoAK47

Comment (zihaoAK47) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: Of course, does the contact information use WeChat, phone number, or Twitter

Comment (strongduanmu) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: I think wechat is better.

Comment (zihaoAK47) [reply parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/DMLStatement.g4:290]: Has been sent to your email

**Pull request merged**